src/overlays/actors/ovl_Boss_Ganon2/z_boss_ganon2.c

```diff
@@ -52,342 +56,3008 @@ const ActorInit Boss_Ganon2_InitVars = {
+            this->unk_3A4.y = this->actor.world.pos.y;
+            this->unk_3A4.z = this->actor.world.pos.z;
+            this->unk_3B0.x = this->actor.world.pos.x;
+            this->unk_3B0.z = this->actor.world.pos.z;
+            this->unk_3B0.y = (this->unk_1B8.y + 60.0f) - 40.0f;
+            if (this->unk_398 > 0xA6 && this->unk_398 < 0xAD) {
+                this->unk_312 = 2;
+            }
+            if (this->unk_398 > 0xBA && this->unk_398 < 0xC4) {
+                this->unk_312 = 1;
+            }
+            if (this->unk_398 > 0xCA && this->unk_398 < 0xD2) {
+                this->unk_312 = 2;
+            }
+            if ((this->unk_398 == 0xA6) || (this->unk_398 == 0xB9) || (this->unk_398 == 0xC8)) {
+                func_80078884(NA_SE_EN_MGANON_SWORD);
+                func_80078884(NA_SE_EN_MGANON_ROAR);
+            }
+            if (this->unk_398 == 0xD7) {
+                this->unk_39C = 0x17;
+                this->unk_224 = 0.0f;
+                func_8002DF54(globalCtx, &this->actor, 0x55);
+            }
+            break;
+        case 23:
+            SkelAnime_Update(&this->skelAnime);
+            if (this->unk_398 > 0xDE && this->unk_398 < 0xE8) {
+                this->unk_312 = 2;
+            }
+            if (this->unk_398 == 0xDE) {
+                func_80078884(NA_SE_EN_MGANON_SWORD);
+                func_80078884(NA_SE_EN_MGANON_ROAR);
+            }
+            this->unk_3A4.x = (player->actor.world.pos.x - 40.0f) + 6.0f;
+            this->unk_3A4.y = player->actor.world.pos.y + 40.0f;
+            this->unk_3A4.z = (player->actor.world.pos.z + 20.0f) - 7.0f;
+            this->unk_3B0.x = player->actor.world.pos.x;
+            this->unk_3B0.y = ((player->actor.world.pos.y + 10.0f + 60.0f) - 20.0f) - 2.0f;
+            this->unk_3B0.z = player->actor.world.pos.z;
+            if (this->unk_398 == 0xE4) {
+                func_80078884(NA_SE_IT_SHIELD_REFLECT_SW);
+                func_8002DF54(globalCtx, &this->actor, 0x56);
+                func_800A9F6C(0.0f, 0xFF, 0xA, 0x32);
+            }
+            if (this->unk_398 >= 0xE5) {
+                globalCtx->envCtx.unk_E1 = 1;
+                globalCtx->envCtx.unk_E2[0] = globalCtx->envCtx.unk_E2[1] = globalCtx->envCtx.unk_E2[2] = 0xFF;
+                globalCtx->envCtx.unk_E2[3] = 0x64;
+                if (this->unk_398 == 0xEA) {
+                    globalCtx->envCtx.unk_E1 = 0;
+                    this->unk_39C = 0x18;
+                    this->unk_398 = 0;
+                    sp68 = player->actor.world.pos;
+                    sp68.y += 60.0f;
+                    func_808FD210(globalCtx, &sp68);
+                    globalCtx->envCtx.unk_D8 = 0.0f;
+                    globalCtx->envCtx.unk_BE = 0;
+                    this->unk_339 = 0;
+                }
+            }
+            break;
+        case 24:
+            SkelAnime_Update(&this->skelAnime);
+            // might be fake?
+            if (1) {}
```

Comment:
dont need to comment on if 1s imo, we can easily search them up if we want to try to get rid of them